tests/operation_balancer/simple_balance_add_tree.mlir

```diff
@@ -0,0 +1,12 @@
+// RUN: heir-opt --operation-balancer %s | FileCheck %s
+
+// CHECK-DAG: %[[ADD_ONE:.*]] = arith.addi %arg0, %arg1
+// CHECK-DAG: %[[ADD_TWO:.*]] = arith.addi %arg2, %arg3
```

Comment:
nit: instead of matching on the full SSA values (the literals "%arg0"), you might want to have a pattern here matching on the func inputs here and above